### PR TITLE
feat(scheduler): clean up stray pods when a release is deleted

### PR DESCRIPTION
**Pull request by helgi**
Scaling an RC to 0 and deleting it does not always clear up all the active pods. This can happen if there is an image pull problem

### CODE REVIEW

File: rootfs/api/models/release.py [hunk 1]
```diff
@@ -183,8 +183,11 @@ def cleanup_old(self):
         )
 
         # Cleanup controllers
+        labels = {
+            'heratige': 'deis'
```
Comment (bacongobbler) [inline rootfs/api/models/release.py]: typo: heritage

File: rootfs/api/models/release.py [hunk 2]
```diff
@@ -208,6 +211,7 @@ def cleanup_old(self):
         # find stray env secrets to remove that may have been missed
         log_event(self.app, 'Cleaning up orphaned environment var secrets', level=logging.DEBUG)
         labels = {
+            'heratige': 'deis',
```
Comment (bacongobbler) [inline rootfs/api/models/release.py]: typo: heritage

File: rootfs/api/models/release.py [hunk 3]
```diff
@@ -220,6 +224,22 @@ def cleanup_old(self):
 
             self._scheduler._delete_secret(self.app.id, secret['metadata']['name'])
 
+        # Remove stray pods
+        labels = {
+            'heratige': 'deis'
```
Comment (bacongobbler) [inline rootfs/api/models/release.py]: typo: heritage

File: rootfs/api/models/release.py [hunk 4]
```diff
@@ -228,13 +248,22 @@ def _delete_release_in_scheduler(self, namespace, version):
         secret that container the env var
         """
         labels = {
+            'heratige': 'deis',
```
Comment (bacongobbler) [inline rootfs/api/models/release.py]: typo: heritage

**Comment by helgi** [comment]
Fypos resolved

**Comment by arschles** [comment]
@helgi any e2e tests that were/should be added for checking stray pods?

**Comment by helgi** [comment]
I can't think of any e2e scenario I can write beyond a very high level "k get po" and filter results that way

**Pull request merged**